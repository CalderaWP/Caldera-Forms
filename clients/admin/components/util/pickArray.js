export const pickArray = (array, key) => {
	return array.reduce(
		(accumualtor, item) =>
			accumualtor.concat([item[key]]), []
	);
}