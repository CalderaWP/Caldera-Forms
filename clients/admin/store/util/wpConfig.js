/**
 * API config and other stuff passed to jsLand via wp_localize_script
 */
export default {
    cfdotcom: 'object' === typeof CF_CLIPPY ? CF_CLIPPY.cfdotcom : {},
    fallback: 'object' === typeof CF_CLIPPY ? CF_CLIPPY.fallback : {},
    extendTitle:  'object' === typeof CF_CLIPPY ? CF_CLIPPY.extend_title : '',
    noForms:  'object' === typeof CF_CLIPPY ? CF_CLIPPY.noForms : '',
};
