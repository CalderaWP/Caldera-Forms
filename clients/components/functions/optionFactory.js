/**
 * Creates option values for select fields
 *
 * @since 1.8.6
 *
 * @param {String|number|Boolean} value Value of option
 * @param {String|number} label Optional. Label for option
 * @param {Object} other Optional. Additional data to add
 * @return {{value: *, label: *}}
 */
export const optionFactory = (value, label = null, other = {} ) => {
	return {
		...other,
		value,
		label: label ? label : value,
	}
};
