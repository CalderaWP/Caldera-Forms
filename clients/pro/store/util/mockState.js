export const mockSTATE = {
	loading: false,
	connected: true,
	forms: [
		{ form_id : 1, name: 'One', layout: 2, pdf_layout:1 },
		{ form_id : 2, name: 'Two', layout: 1, pdf_layout:2 },
	],
	settings : {
		enhancedDelivery: true,
		generatePDFs: false
	},
	layouts : [
		{ id: 1, name: 'One' },
		{ id: 2, name: 'Two' }
	],
	account: {
		plan: 'apex',
		id: 42,
		apiKeys: {
			public: 'public',
			secret: 'secret',
			token: 'token'
		}
	}

};