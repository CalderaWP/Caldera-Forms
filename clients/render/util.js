const SparkMD5 = require('spark-md5');

export const getFieldConfigBy = (fieldConfigs, findBy, findWhere) => {
	return fieldConfigs.find(field => findWhere === field[findBy]);

};


export const hashFile = (file, callback) => {
	let blobSlice = File.prototype.slice || File.prototype.mozSlice || File.prototype.webkitSlice,
		chunkSize = 2097152,                             // Read in chunks of 2MB
		chunks = Math.ceil(file.size / chunkSize),
		currentChunk = 0,
		spark = new SparkMD5.ArrayBuffer(),
		fileReader = new FileReader();

	fileReader.onload = function (e) {
		spark.append(e.target.result);                   // Append array buffer
		currentChunk++;

		if (currentChunk < chunks) {
			loadNext();
		} else {
			callback(spark.end());
		}
	};

	fileReader.onerror = function () {
		console.warn('oops, something went wrong.');
	};

	function loadNext() {
		var start = currentChunk * chunkSize,
			end = ((start + chunkSize) >= file.size) ? file.size : start + chunkSize;

		fileReader.readAsArrayBuffer(blobSlice.call(file, start, end));
	}

	loadNext();
}



export const removeFromPending = (fieldId, cf2) => {
	const index = cf2.pending.findIndex(item => item === fieldId);
	if (-1 < index) {
		cf2.pending.splice(index, 1);
	}
}

export const removeFromUploadStarted = (fieldId, cf2) => {
	const index = cf2.uploadStarted.findIndex(item => item === fieldId);
	if (-1 < index) {
		cf2.uploadStarted.splice(index, 1);
	}
}

export const removeFromBlocking = (fieldId, cf2, fieldConfig = {}) => {
	const index = cf2.fieldsBlocking.findIndex(item => item === fieldId);
	if (-1 < index) {
		cf2.fieldsBlocking.splice(index, 1);
	}
	setSubmitButtonState(cf2, fieldConfig);
}

export const setBlocking = ( fieldId, cf2, fieldConfig = {}) => {
	removeFromUploadStarted(fieldId, cf2);
	removeFromPending(fieldId, cf2);
	if(cf2.fieldsBlocking.indexOf(fieldId) < 0){
		cf2.fieldsBlocking.push( fieldId );
	}
	setSubmitButtonState(cf2, fieldConfig, false);
}

/**
 * Default process to convert a file to media
 *
 * @param file
 * @param additionalData
 * @param fetch
 * @return {*}
 */
export const createMediaFromFile = (file, additionalData, fetch) => {

	// Create upload payload
	const data = new window.FormData();
	data.append('file', file, file.name || file.type.replace('/', '.'));
	data.append('title', file.name ? file.name.replace(/\.[^.]+$/, '') : file.type.replace('/', '.'));
	Object.keys(additionalData)
		.forEach(key => data.append(key, additionalData[key]));

	return fetch(additionalData.API_FOR_FILES_URL, {
		body: data,
		method: 'POST',
		headers: {
			'X-WP-Nonce': additionalData._wp_nonce
		}
	});
}

/**

 * Converts the bytes values to smaller value in corresponding unit
 *
 * @param bytes value to be converted
 * @return {*}
 */
export const sizeFormat = (bytes) => {
	let converted = false;
	const quant = [
		{
			unit: 'TB',
			mag: 1099511627776
		},

		{
			unit: 'GB',
			mag: 1073741824
		},

		{
			unit: 'MB',
			mag: 1048576
		},

		{
			unit: 'kB',
			mag: 1024
		},

		{
			unit: 'B ',
			mag: 1
		}
	];
	quant.forEach(function(v){
		if (parseFloat(bytes) >= v.mag && converted == false){
			converted = bytes/v.mag;
			if( bytes > 1048576 ){
				converted = converted.toFixed(2);
			}else{
				converted = Math.round( converted );
			}
			converted = converted +' '+v.unit;
		}
	});
	return converted;
}

/* Set Submit button state
 *
 * @param {object} cf2
 * @param {object} fieldConfig
 * @param {boolean} state
 * @return {*}
 */
export const setSubmitButtonState = (cf2, fieldConfig, state) => {

	const fieldIdAttr = fieldConfig.fieldIdAttr;

	const formIdAttr = getFormIdAttr(cf2, fieldIdAttr);

	const form =  jQuery("#" + formIdAttr);
	//If no state param was send in the function define the state based on elements inside the fieldsBlocking Array
	if(typeof state === "undefined") {
		state = cf2.fieldsBlocking.length <= 0;
	}

	//If state === true enable submit button else disable submit button
	if(state) {
		form.find(':submit').prop('disabled',false);
	} else {
		form.find(':submit').prop('disabled',true);
	}
	return state;
}

/**
 * Find the formIdAttr corresponding to a fieldIdAttr in the cf2 object
 *
 * @param {object} cf2
 * @param {string} fieldIdAttr
 * @return {string} formIdAttr
 */
export const getFormIdAttr = (cf2, fieldIdAttr) => {

	let formIdAttr = cf2.formIdAttr;

	if( typeof formIdAttr === "undefined" ){

		const entries = Object.entries(cf2);
		const formEntries = [];
		entries.forEach( entry => {
			if(entry[1].hasOwnProperty("fields") ) {
				formEntries.push(entry);
			}
		})
		formEntries.forEach( formEntry => {
			if(formEntry[1].fields.hasOwnProperty(fieldIdAttr ) ) {
				formIdAttr = formEntry[0];
			}
		})
	}

	return formIdAttr;

}

/**
 * Start or Stop spinner animation to indicate process is happening
 *
 * @since 1.8.0
 *
 * @param {string} formIdAttr Id Attribute of the <form> tag
 * @param {boolean} trigger if true force addClass if false force removeClass
 */
export const processAnimation = (formIdAttr, trigger) => {

	const cfElement = formIdAttr.length > 0 ? jQuery("#" + formIdAttr).parent(".caldera-grid") : false;

	if(cfElement !== false) {
		if(trigger === false){
			if(jQuery(cfElement).hasClass("cf_processing")){
				jQuery(cfElement).removeClass("cf_processing");
			}
		}else if(trigger === true){
			if(!jQuery(cfElement).hasClass("cf_processing")){
				jQuery(cfElement).addClass("cf_processing");
			}
		}
	}

};

/**
 * Add a component -- CalderaFormsRender -- to window.cf2 correctly
 *
 * @since 1.8.0
 *
 * @param {function} component
 * @param {string}idAttr
 * @param {object}window
 */
export const captureRenderComponentRef = (component,idAttr,window = {}) => {
	window.cf2 = window.cf2 || {};
	window.cf2[idAttr] = {
		...window.cf2[idAttr] || {},
		component
	};
};
