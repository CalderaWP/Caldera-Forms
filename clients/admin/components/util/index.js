import {optionFactory} from "./optionFactory";
import {pickArray} from "./pickArray";
import {settingsProp} from "../../state/prepareSettings";

export default {
	optionFactory,
	pickArray,
	settingsProp
}