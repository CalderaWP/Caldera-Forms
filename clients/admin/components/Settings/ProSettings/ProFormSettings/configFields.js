import {optionFactory} from "../../../util/optionFactory";

export const PRO_FORM_SEND_LOCAL = 'SETTINGS/PRO/FORM/SEND_LOCAL';
export const PRO_FORM_EMAIL_LAYOUT = 'SETTINGS/PRO/FORM/EMAIL_LAYOUT';
export const PRO_FORM_PDF_LAYOUT = 'SETTINGS/PRO/FORM/PDF_LAYOUT';
export const PRO_FORM_PDF_ATATCH = 'SETTINGS/PRO/FORM/PDF_ATTACH ';
export const PRO_FORM_PDF_LINK = 'SETTINGS/PRO/FORM/PDF__LINK';

const enableOption = optionFactory(
	true,
	'Enable'
);

const disableOption = optionFactory(
	false,
	'Disable'
);

const enableDiable = [enableOption,disableOption]
export default [
	{
		id: PRO_FORM_SEND_LOCAL,
		label: 'Disable enhanced delivery for this form',
		type: 'checkbox',
		path: 'pro.send_local',
		options: [enableDiable]
	},
	{
		id: PRO_FORM_EMAIL_LAYOUT,
		label: 'Email Layout',
		type: 'dropdown',
		path: 'pro.layout',
		options: []
	},
	{
		id: PRO_FORM_PDF_LAYOUT,
		label: 'PDF Layout',
		type: 'dropdown',
		path: 'pro.pdf_layout',
	},
	{
		id: PRO_FORM_PDF_ATATCH,
		label: 'Attach PDF To Main Mailer',
		type: 'checkbox',
		path: 'pro.attach_pdf',
		options: [enableDiable]
	},
	{
		id: PRO_FORM_PDF_LINK,
		label: 'Add PDF Link',
		type: 'checkbox',
		path: 'pro.pdf_link',
		options: [enableDiable]
	},
]