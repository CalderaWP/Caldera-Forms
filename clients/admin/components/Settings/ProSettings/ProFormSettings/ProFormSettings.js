import React from 'react';
import propTypes from 'prop-types';
import {SettingsGroup} from "../../SettingsGroup";
import configFields, {PRO_FORM_EMAIL_LAYOUT, PRO_FORM_PDF_LAYOUT} from './configFields'
/**
 * Create the ProFormSettings UI
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export class ProFormSettings extends SettingsGroup {

	/**
	 * Get config fields
	 *
	 * Adds pro layouts as options when possible
	 * @return {Array}
	 */
	getConfigFields(){
		let configFields = super.getConfigFields();
		if ( this.props.layouts.length ) {
			configFields.forEach(configField => {
				if ([
					PRO_FORM_EMAIL_LAYOUT,
					PRO_FORM_PDF_LAYOUT
				].includes(configField.id)) {
					configField.options = this.props.layouts;
				}
			})
		}
		return configFields;
	}
};

/**
 * Prop types for the ProFormSettings component
 * @type {{}}
 */
ProFormSettings.propTypes = {
	classNames: propTypes.string,
	layouts: propTypes.array
};

/**
 * Default props for the ProFormSettings component
 * @type {{}}
 */
ProFormSettings.defaultProps = {
	configFields,
	wrapperClass: 'caldera-forms-pro-form-settings',
	layouts: []
};
