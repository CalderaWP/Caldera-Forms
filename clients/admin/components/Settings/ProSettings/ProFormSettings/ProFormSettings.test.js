import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ProFormSettings} from "./ProFormSettings";
import {optionFactory} from "../../../util/optionFactory";

Enzyme.configure({ adapter: new Adapter() });

describe( 'ProFormSettings component', () => {
	it( 'Matches snapshot with minimal props', () => {
		expect(
			renderer.create(
				<ProFormSettings/>
			).toJSON()
		).toMatchSnapshot()
	});


	it( 'Matches snapshot with layouts', () => {
		expect(
			renderer.create(
				<ProFormSettings
					layouts={[
						optionFactory(1,"one"),
						optionFactory(2,'two')

					]}

				/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Is wrapped in the right class', () => {
		expect(
			shallow(
				<ProFormSettings/>
			).find( '.' + ProFormSettings.defaultProps.wrapperClass )
				.length
		).toEqual(1)
	});
});