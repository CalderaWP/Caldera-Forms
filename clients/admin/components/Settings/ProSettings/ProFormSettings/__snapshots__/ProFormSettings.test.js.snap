// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProFormSettings component Matches snapshot with layouts 1`] = `
<div
  className="caldera-forms-pro-form-settings"
>
  <div
    className="caldera-config-field-setup"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/SEND_LOCAL"
        >
          Disable enhanced delivery for this form
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/SEND_LOCAL"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/EMAIL_LAYOUT"
        >
          Email Layout
        </label>
        <select
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/EMAIL_LAYOUT"
          onChange={[Function]}
          value={undefined}
        >
          <option
            className="caldera-config-option"
            value={1}
          >
            one
          </option>
          <option
            className="caldera-config-option"
            value={2}
          >
            two
          </option>
        </select>
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF_LAYOUT"
        >
          PDF Layout
        </label>
        <select
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF_LAYOUT"
          onChange={[Function]}
          value={undefined}
        >
          <option
            className="caldera-config-option"
            value={1}
          >
            one
          </option>
          <option
            className="caldera-config-option"
            value={2}
          >
            two
          </option>
        </select>
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF_ATTACH "
        >
          Attach PDF To Main Mailer
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF_ATTACH "
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF__LINK"
        >
          Add PDF Link
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF__LINK"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value={undefined}
        />
      </div>
    </div>
  </div>
  <button
    autoFocus={undefined}
    className="components-button"
    disabled={undefined}
    onClick={undefined}
    type="button"
  />
</div>
`;

exports[`ProFormSettings component Matches snapshot with minimal props 1`] = `
<div
  className="caldera-forms-pro-form-settings"
>
  <div
    className="caldera-config-field-setup"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/SEND_LOCAL"
        >
          Disable enhanced delivery for this form
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          defaultChecked={undefined}
          disabled={false}
          id="SETTINGS/PRO/FORM/SEND_LOCAL"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/EMAIL_LAYOUT"
        >
          Email Layout
        </label>
        <select
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/EMAIL_LAYOUT"
          onChange={[Function]}
          value={undefined}
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF_LAYOUT"
        >
          PDF Layout
        </label>
        <select
          className="field-config field-config"
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF_LAYOUT"
          onChange={[Function]}
          value={undefined}
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF_ATTACH "
        >
          Attach PDF To Main Mailer
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          defaultChecked={undefined}
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF_ATTACH "
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="SETTINGS/PRO/FORM/PDF__LINK"
        >
          Add PDF Link
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          defaultChecked={undefined}
          disabled={false}
          id="SETTINGS/PRO/FORM/PDF__LINK"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="checkbox"
        />
      </div>
    </div>
  </div>
  <button
    autoFocus={undefined}
    className="components-button"
    disabled={undefined}
    onClick={undefined}
    type="button"
  />
</div>
`;
